dot/network/stream_manager.go

```diff
@@ -0,0 +1,68 @@
+package network
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/network"
+)
+
+var cleanupStreamInterval = time.Minute
+
+// streamManager tracks inbound streams and runs a cleanup goroutine every `cleanupStreamInterval` to close streams that
+// we haven't received any data on for the last time period. this prevents keeping stale streams open and continuously trying to
+// read from it, which takes up lots of CPU over time.
+type streamManager struct {
+	ctx                 context.Context
+	lastReceivedMessage *sync.Map //map[string]time.Time
```

Comment:
Why are we using two maps?
We can have a single map with the following value
```
type val struct {
  s network.Stream
  t time.time.
}
```

Reply:
updated!